tests/Aspire.Hosting.Azure.Tests/AzureDeployerTests.cs

```diff
@@ -203,6 +203,57 @@ public async Task DeployAsync_WithContainer_Works()
+        var deploymentOutputs = new Dictionary<string, object>
+        {
+            ["env_AZURE_CONTAINER_REGISTRY_NAME"] = new { type = "String", value = "testregistry" },
+            ["env_AZURE_CONTAINER_REGISTRY_ENDPOINT"] = new { type = "String", value = "testregistry.azurecr.io" },
+            ["env_AZURE_CONTAINER_REGISTRY_MANAGED_IDENTITY_ID"] = new { type = "String", value = "/subscriptions/test/resourceGroups/test-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/test-identity" },
+            ["env_AZURE_CONTAINER_APPS_ENVIRONMENT_DEFAULT_DOMAIN"] = new { type = "String", value = "test.westus.azurecontainerapps.io" },
+            ["env_AZURE_CONTAINER_APPS_ENVIRONMENT_ID"] = new { type = "String", value = "/subscriptions/test/resourceGroups/test-rg/providers/Microsoft.App/managedEnvironments/testenv" }
+        };
+        var armClientProvider = new TestArmClientProvider(deploymentOutputs);
+        ConfigureTestServices(builder, armClientProvider: armClientProvider, processRunner: mockProcessRunner);
+
+        var containerAppEnv = builder.AddAzureContainerAppEnvironment("env");
+        var azureEnv = builder.AddAzureEnvironment();
+        builder.AddDockerfile("api", "api.Dockerfile");
+
+        // Act
+        using var app = builder.Build();
+        await app.StartAsync();
+        await app.WaitForShutdownAsync();
+
+        // Assert that container environment outputs are propagated to outputs because they are
+        // hoisted up for the container resource
+        Assert.Equal("testregistry", containerAppEnv.Resource.Outputs["AZURE_CONTAINER_REGISTRY_NAME"]);
+        Assert.Equal("testregistry.azurecr.io", containerAppEnv.Resource.Outputs["AZURE_CONTAINER_REGISTRY_ENDPOINT"]);
+        Assert.Equal("/subscriptions/test/resourceGroups/test-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/test-identity", containerAppEnv.Resource.Outputs["AZURE_CONTAINER_REGISTRY_MANAGED_IDENTITY_ID"]);
+        Assert.Equal("test.westus.azurecontainerapps.io", containerAppEnv.Resource.Outputs["AZURE_CONTAINER_APPS_ENVIRONMENT_DEFAULT_DOMAIN"]);
+        Assert.Equal("/subscriptions/test/resourceGroups/test-rg/providers/Microsoft.App/managedEnvironments/testenv", containerAppEnv.Resource.Outputs["AZURE_CONTAINER_APPS_ENVIRONMENT_ID"]);
+
+        // Assert - Verify ACR login command was called since Dockerfile image needs to be pushed
+        Assert.Contains(mockProcessRunner.ExecutedCommands,
+            cmd => cmd.ExecutablePath.Contains("az") &&
+                   cmd.Arguments == "acr login --name testregistry");
+
+        // Assert - Verify Docker tag and push called for Dockerfile
+        Assert.Contains(mockProcessRunner.ExecutedCommands,
+            cmd => cmd.ExecutablePath == "docker" &&
+                   cmd.Arguments != null &&
+                   cmd.Arguments.StartsWith("tag api testregistry.azurecr.io/"));
```

Comment:
What's up with the slash

Reply:
The tag command includes the image name so the full command would be something like `tag api testregistry.azurecr.io/api:latest`. We could probably assert on the entire thing, I can look at doing that in one of the other PRs.